contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/AbstractHttpHandler.java

```diff
@@ -0,0 +1,204 @@
+    checkNotNull(context, "context is required");
+    context.sentMessageSize.addAndGet(bytes);
+    recordMessageEvent(
+        context.span,
+        context.sentSeqId.addAndGet(1L),
+        HttpTraceConstants.MSG_EVENT_ATTR_SENT,
+        bytes,
+        0L);
+  }
+
+  /**
+   * Instrument an HTTP span after a message is received. Typically called for every chunk of
+   * request or response is received.
+   *
+   * @param context request specific {@link HttpRequestContext}
+   * @param bytes bytes received.
+   */
+  public final void handleMessageReceived(HttpRequestContext context, long bytes) {
+    checkNotNull(context, "context is required");
+    context.receiveMessageSize.addAndGet(bytes);
+    recordMessageEvent(
+        context.span,
+        context.receviedSeqId.addAndGet(1L),
+        HttpTraceConstants.MSG_EVENT_ATTR_RECEIVED,
+        bytes,
+        0L);
+  }
+
+  void spanEnd(Span span, int httpStatus, @Nullable Throwable error) {
+    span.setAttribute(
+        HttpTraceConstants.HTTP_STATUS_CODE, AttributeValue.longAttributeValue(httpStatus));
+    if (error != null) {
+      String message = extractErrorMessage(error);
+      span.setAttribute(
+          HttpTraceConstants.HTTP_STATUS_TEXT, AttributeValue.stringAttributeValue(message));
+      getLogger().log(Level.FINE, message, error);
+    }
+    span.setStatus(statusConverter.convert(httpStatus));
+    span.end();
+  }
+
+  final String getSpanName(Q request) {
+    String spanName = extractor.getRoute(request);
+    if (spanName == null) {
+      spanName = "/";
+    }
+    if (!spanName.startsWith("/")) {
+      spanName = "/" + spanName;
+    }
+    return spanName;
+  }
+
+  final void addSpanRequestAttributes(Span span, Q request) {
+    span.setAttribute(HttpTraceConstants.COMPONENT, HttpTraceConstants.COMPONENT_ATTR_VALUE);
+    putAttributeIfNotEmptyOrNull(
+        span, HttpTraceConstants.HTTP_METHOD, extractor.getMethod(request));
+    putAttributeIfNotEmptyOrNull(span, HttpTraceConstants.HTTP_ROUTE, extractor.getRoute(request));
+    putAttributeIfNotEmptyOrNull(span, HttpTraceConstants.HTTP_URL, extractor.getUrl(request));
+    putAttributeIfNotEmptyOrNull(
+        span, HttpTraceConstants.HTTP_USERAGENT, extractor.getUserAgent(request));
+    putAttributeIfNotEmptyOrNull(
+        span, HttpTraceConstants.HTTP_FLAVOR, extractor.getHttpFlavor(request));
+    putAttributeIfNotEmptyOrNull(
+        span, HttpTraceConstants.HTTP_CLIENTIP, extractor.getClientIp(request));
+  }
+
+  String extractErrorMessage(Throwable error) {
+    String message = error.getMessage();
+    if (message == null) {
+      message = error.getClass().getSimpleName();
+    }
+    return message;
+  }
+
+  protected void recordMeasurements(HttpRequestContext context, int httpCode) {
+    try {
+      meter.newMeasureBatchRecorder();
+      meter.longCounterBuilder(HttpTraceConstants.MEASURE_COUNT + httpCode).build();
+    } catch (UnsupportedOperationException ignore) {
```

Comment:
I assume this is temporary? If so, can you add a TODO and create an issue to track it?